scripts/where/index.ts

```diff
@@ -0,0 +1,162 @@
+import * as fs from 'fs';
+import * as glob from 'glob';
+import * as path from 'path';
+import { findGitRoot, fetchRemote, git, getPackageInfos } from 'workspace-tools';
+import { rollup as lernaAlias } from 'lerna-alias';
+
+interface PackageNameInfo {
+  folder7: string;
+  folder8: string;
+  name7: string;
+  name8: string;
+  componentNames7: string[];
+  componentNames8: string[];
+}
+
+interface ComponentLocationInfo {
+  name: string;
+  folder7: string;
+  folder8: string;
+  examples7?: string;
+  examples8?: string;
+}
+
+function lsTree(branch: string, ...args: string[]): string[] {
+  const result = git(['ls-tree', '--name-only', branch, ...args], { cwd });
+  if (!result.success) {
+    console.error('git ls-tree failed:');
+    console.error(result.stderr);
+    process.exit(1);
+  }
+  return result.stdout.split(/\r?\n/g);
+}
+
+function getUnscopedName(pkg: string) {
+  return pkg.includes('@') ? pkg.split('/')[1] : pkg;
+}
+
+const lastSharedCommit = '56856bb94c';
+
+const packageNames8to7 = {
+  '@fluentui/azure-themes': '@uifabric/azure-themes',
+  '@fluentui/react': 'office-ui-fabric-react',
+  '@fluentui/react-charting': '@uifabric/charting',
+};
+
+const packageNames7to8 = {
+  '@fluentui/react': 'office-ui-fabric-react',
+};
+
+const packagesInApps = ['public-docsite', 'public-docsite-resources', 'fabric-website', 'fabric-website-resources'];
+
+const cwd = process.cwd();
+const gitRoot = findGitRoot(cwd);
+/** Whether the current branch is based on 7 (probably) */
+const is7 = fs.existsSync(path.join(gitRoot, 'packages/office-ui-fabric-react/package.json'));
+
+fetchRemote('upstream', cwd);
+
+function whereIsComponent(componentName: string, packageName: string) {}
+
+const componentsGlob = '{apps,packages}/!(react-docsite-components)/src/components/!(*.*)';
+const packageJsonGlob = '{apps,packages}/!(fluentui|web-components)/package.json';
+
+const components7 = lsTree('upstream/7.0', componentsGlob);
+const components8 = lsTree('upstream/master', componentsGlob);
+
+const componentFolders = glob.sync('packages/!(react-docsite-components)/src/components/!(*.*)', { cwd: gitRoot });
+
+const packages = lernaAlias({ sourceDirectory: false });
+
+function whereIsPackage(packageName: string) {
```

Comment:
Nice :) but i don't think this is intended for this PR?

Reply:
Oops 😆 will remove